inst/tests/tests.Rraw

```diff
@@ -18059,6 +18059,13 @@ for (opt in c(0,Inf)) {
   testnum = 2233.44
 }
 options(old)
+# optimized := with gforce functions that can return lists #5403
+DT = data.table(grp=1:2, x=1:4)
+test(2233.45, DT[, `:=`(y = shift(x, type = "lag", n = 1), z = shift(x = x, type = "lead", n = 1)), by = grp], data.table(grp=1:2, x=1:4, y=c(NA,NA,1:2), z=c(3:4, NA, NA)))
+DT = data.table(grp = 1:2, x=1:4)
+test(2233.46, DT[, l := shift(x, n=c(0, 0)), by=grp], data.table(grp=1:2, x=1:4, l=lapply(1:4, rep, 2)))
```

Comment:
any reason to use functional form `` `:=`(...)`` above, but operator form `x := y` here? Especially in the next test with length-2 LHS.

Reply:
I guess I find the functional form more intuitive than wrapping the result inside a list when using multiple assignments. `shift` is a here a somewhat special case since it already returns a list.

Reply:
got it, I see the difference now. both of the latter tests have only one call on RHS.